pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -787,12 +787,7 @@ public CompletableFuture<Boolean> allowTopicOperationAsync(TopicName topicName,
             return CompletableFuture.completedFuture(false);
         }
         if (isProxyRole(role)) {
-            CompletableFuture<Boolean> isRoleAuthorizedFuture = allowTopicOperationAsync(
-                    topicName, operation, role, authData);
-            CompletableFuture<Boolean> isOriginalAuthorizedFuture = allowTopicOperationAsync(
-                    topicName, operation, originalRole, authData);
-            return isRoleAuthorizedFuture.thenCombine(isOriginalAuthorizedFuture,
-                    (isRoleAuthorized, isOriginalAuthorized) -> isRoleAuthorized && isOriginalAuthorized);
+            return allowTopicOperationAsync(topicName, operation, originalRole, authData);
```

Comment:
I think that the original change was made by @michaeljmarshall together with other changes with the intention to fix a CVE, possibly https://pulsar.apache.org/security/CVE-2023-30429/. 
@michaeljmarshall do you have a chance to review?

Reply:
It's not clear to me why this PR only proposes changing this block of code. All 9 times we use the `isProxyRole` method in the `AuthorizationService` class, we validate that both the proxy's role and the `originalRole` have permission to perform a given action. @rdhabalia - why would we only modify this one?

Regarding the change itself, it was introduced in https://github.com/apache/pulsar/pull/7788. I remember reading somewhere that the idea was you could have dedicated proxy's for a given set of clients and then you could choose more granular permissions on a per proxy basis.

If you want the proxy to be a super user and want to functionally skip the permission check, just add it to the super user list.